service/src/main/java/bio/terra/workspace/service/workspace/flight/CheckSpendProfileStep.java

```diff
@@ -45,11 +56,25 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
             .getSpendProfileId()
             .orElseThrow(() -> MissingSpendProfileException.forWorkspace(workspaceUuid));
 
-    SpendProfile spendProfile = spendProfileService.authorizeLinking(spendProfileId, userRequest);
-    if (spendProfile.billingAccountId().isEmpty()) {
-      throw NoBillingAccountException.forSpendProfile(spendProfileId);
+    SpendProfile spendProfile =
```

Comment:
I think there are errors from `authorizeLinking` that should be retried. For example, network or service unavailable kind of errors that show up during service upgrade. A single `BillingProfileManagerServiceApiException` isn't sufficient to distinguish the retry-able cases from the fatal cases.

Suggest distinguishing between those cases in the service and adding retry to the uses of this step.